reference/errors/error_codes.md

```diff
@@ -154,9 +158,9 @@ The `minWordSizeForTypos` object is invalid. The value for both `oneTypo` and `t
 
 The field you are trying to modify is immutable.
 
-## `primary_key_inference_failed`
+## `index_primary_key_no_candidate_found`
 
-The first provided document contains no fields with the substring `id`. [Manually designate the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key) or add `id` to one of your fields so it can be used as the primary key during inference. We recommend manually setting the primary key.
+[Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the index does not contain any fields ending with `id`. [Manually designate the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key) or add `id` to one of your fields so it can be used as the primary key during inference. We recommend manually setting the primary key.
```

Comment:
I don't think we should make this recommendation, as their dataset might not contain a field suitable to be used as a primary key (i.e. present in every document and containing unique values).

```suggestion
[Primary key inference](/learn/core_concepts/primary_key.md#meilisearch-guesses-your-primary-key) failed as the received documents do not contain any fields ending with `id`. [Manually designate the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key), or add some field ending with `id` to your documents.
```